osmnx/utils_graph.py

```diff
@@ -129,6 +136,20 @@ def graph_from_gdfs(gdf_nodes, gdf_edges, graph_attrs=None):
     -------
     G : networkx.MultiDiGraph
     """
+    if not ("x" in gdf_nodes.columns and "y" in gdf_nodes.columns):
+        raise ValueError("gdf_nodes must contain x and y columns")
+
+    # if gdf_nodes has a geometry attribute set, drop that column (as we use x
+    # and y for geometry information) and warn the user if the geometry coords
+    # differ from those in the x and y columns
+    if hasattr(gdf_nodes, "geometry"):
+        if (gdf_nodes.geometry.x != gdf_nodes["x"]).all() or (
```

Comment:
If for some reason the geometry column contains something other than `Point` objects, access to `geometry.x` will raise
```python
ValueError: x attribute access only provided for Point geometries
```
Do you want to add your own check that every node geometry is indeed a Point, or maybe update the documentation to say that the geometry column in the nodes data frame will be discarded, and further will throw an error if it contains anything other than single points?

Reply:
Thanks. I see that ValueError as an example of "the geometry values do not match the x and y columns' values". Rather than handle it separately (or just let it error), I've revised the PR in ae3e4b2 to show the warning if it occurs. I think this clearer warning + documentation should help users engaging in an edge case adjust appropriately.  The key thing here would be prevent "silent" weird behavior, as you documented in #677: as long as there's a warning (or at least an error) instead of silently corrupting the data structures.

Reply:
Thanks! Looks good to me 👍